### PR TITLE
fix: handling reseting transport mode and line. 

### PR DESCRIPTION
**Pull request by jonasbrunvoll**
This PR adds `setTransportModeAndResetLineAndStops` `setLineAndResetStops`, two functions used to  correctly handle `line`, `fromStop` and `toStop` whenever `transportMode` and/or `line` is set.

### CODE REVIEW

File: src/page-modules/contact/ticket-control/forms/feedbackForm.tsx [hunk 1]
```diff
@@ -20,6 +21,10 @@ export const FeedbackForm = ({ state, send }: FeedbackFormProps) => {
   const { t } = useTranslation();
   const { getLinesByMode, getQuaysByLine } = useLines();
 
+  useEffect(() => {
+    console.log('state.context.line: ', state.context.line);
+  }, [state]);
+
```
Comment (mortennordseth) [inline src/page-modules/contact/ticket-control/forms/feedbackForm.tsx:27]:
```suggestion
```

File: src/page-modules/contact/ticket-control/forms/feedbackForm.tsx [hunk 2]
```diff
@@ -10,6 +10,7 @@ import { Typo } from '@atb/components/typography';
 import { FileInput } from '../../components/input/file';
 import { ticketControlFormEvents } from '../events';
 import { ContextProps } from '../ticket-control-form-machine';
+import { useEffect } from 'react';
```
Comment (mortennordseth) [inline src/page-modules/contact/ticket-control/forms/feedbackForm.tsx:13]:
```suggestion
```

File: src/page-modules/contact/ticket-control/ticket-control-form-machine.ts [hunk 3]
```diff
@@ -88,7 +90,7 @@ export type ContextProps = {
 };
 
 // Function to reset the agreement fields and error messages
-const resetAgreementFieldsAndErrors = (
+const setFormtypeAndInitialContext = (
```
Comment (mortennordseth) [inline src/page-modules/contact/ticket-control/ticket-control-form-machine.ts:93]:
```suggestion
const setFormTypeAndInitialContext = (
```

File: src/page-modules/contact/means-of-transport/means-of-transport-form-machine.ts [hunk 4]
```diff
@@ -77,6 +79,12 @@ export const meansOfTransportFormMachine = setup({
         } else {
           context.errorMessages[inputName] = [];
         }
+
+        if (inputName === 'transportMode')
+          return setTransportModeAndResetLineAndStops(context, value);
+
+        if (inputName === 'line') return setLineAndResetStops(context, value);
+
```
Comment (mortennordseth) [inline src/page-modules/contact/means-of-transport/means-of-transport-form-machine.ts:87]: I think that we should create separate functions when we have special cases like this. That is more explicit in my opinion and also easier to maintain.

Comment (jonasbrunvoll) [reply src/page-modules/contact/means-of-transport/means-of-transport-form-machine.ts]: Moved both `setTransportModeAndResetLineAndStops ` and `setLineAndResetStops ` to their own actions✅

File: src/page-modules/contact/commoneEvents.ts [hunk 5]
```diff
@@ -12,14 +15,22 @@ export const commonEvents = {} as
...
         | 'SWIFT'
         | 'toStop'
-        | 'transportMode'
         | 'fromStop'
         | 'plannedDepartureTime';
       value: any;
+    }
+  | {
+      type: 'ON_TRANSPORTMODE_CHANGE';
+      inputName: 'transportMode';
```
Comment (mortennordseth) [inline src/page-modules/contact/commoneEvents.ts:29]: I don't think `inputName` is necessary

File: src/page-modules/contact/means-of-transport/means-of-transport-form-machine.ts [hunk 6]
```diff
@@ -77,13 +79,31 @@ export const meansOfTransportFormMachine = setup({
...
           ...context,
           [inputName]: value,
         };
       }
       return context;
     }),
+    onTransportModeChange: assign(({ context, event }) => {
+      if (event.type === 'ON_TRANSPORTMODE_CHANGE') {
+        const { inputName, value } = event;
+        if (inputName === 'transportMode')
```
Comment (mortennordseth) [inline src/page-modules/contact/means-of-transport/means-of-transport-form-machine.ts:93]: `onTransportModeChange` should only be used by inputs handling transportMode. You shouldn't need to check the input name here.

**Review by mortennordseth — APPROVED** [review]

**Pull request merged**